src/canonicalize/CIR.zig

```diff
@@ -274,19 +274,234 @@ pub const PatternRecordField = struct {
     pub const Span = struct { start: u32, len: u32 };
 };
 
+/// Represents an arbitrary precision smallish decimal value
+pub const SmallDecValue = struct {
+    numerator: i16,
+    denominator_power_of_ten: u8,
+
+    /// Convert a small dec to f64 (use for size comparisons)
+    /// TODO: Review, claude generated
```

Comment:
We could just reuse the existing `SmallDec` implementation, yeah?

Reply:
not sure – possibly! can you direct me to where that is? when i search for `SmallDec` in code base, i don't see it!

Reply:
It used to be in `Can`, so I guess this *is* the `SmallDec` implementation now. 😆